custom-accent-colors@demiskp/extension.js

```diff
@@ -203,4 +209,34 @@ export default class CustomAccentColors extends Extension {
             this.deleteFileDir(shellThemeDir.get_path());
         }
     }
+
+    updateIconTheming(apply) {
+        const iconTheme = new Gio.Settings({
+            schema: 'org.gnome.desktop.interface',
+        });
+        const meDir = this.path;
+        const homeDir = GLib.get_home_dir();
+        let iconThemeDir = Gio.File.new_for_path(homeDir + '/.local/share/icons/');
+        let iconThemeDirText = iconThemeDir.get_path()
+        if (apply && this.accentColor != 'default' && this.accentColor != 'blue') {
+            if (!iconThemeDir.query_exists(null)) {
+                this.createDir(iconThemeDir.get_path());
+            }
+            let iconThemeTar = Gio.File.new_for_path(
+                meDir +
+                '/resources/' +
+                this.accentColor + '/' +
+                'custom-icon-colors.tar.gz'
+            );
+            let iconThemeTarText = iconThemeTar.get_path()
+            GLib.spawn_command_line_sync(
```

Comment:
It's advisable to not have GNOME extensions spawn commands. I think it would be better to have the icon dirs uncompressed in the "resources" dir and simply copy them from there to the appropriate dir, as I've done with the shell themes.

Reply:
Ah I was wondering about that. Problem is I don't think I can copy a folder at once. I will certainly start work on a file by file copy system, but if you know any way to copy a folder or use a GLib / Gio way of moving or extracting files I would love to hear it, I just haven't ran across any.